### PR TITLE
Py deps upgrade, TF upgrade, TF fixes

### LINKED ISSUE #145: Deploy errors using the IAM Group
## Background

Deployments by AWS users in the IAM Group do not appear to be working.

Detailed below are the setup steps I went through while debugging this.

Am I missing something?

## Steps to Reproduce

I downloaded the IAM Group Terraform file from the [setup documentation] locally and replaced the `account` and `prefix` variable values then created the IAM Group with Terraform. I then created a new deploy user, added it to the group, and ran `./manage.py deploy`.

The first deployment returned two "…not authorized to perform kms:TagResource" errors. I saw the comment about adding the generated KMS keys to the IAM Group Terraform file but it seemed a little strange to need to go back and modify that file after the fact with any hard-coded values so I added the `kms:TagResource` action to the relevant statement and updated the IAM Group's policy with Terraform. Deploying again returned a "MalformedPolicyDocumentException" error which appears to be a [known issue] with this Terraform provider? Running the deploy step again got past that error but then returned a "…not authorized to perform lambda:CreateEventSourceMapping" error at which point I ended up giving up and deploying with an admin account.

## Desired Change

The least-privilege permissions Terraform file creates a working policy for deploy users.

[setup]: http://binaryalert.io/iam-group.html
[known issue]: https://github.com/hashicorp/terraform/issues/6576

### LINKED ISSUE #166: Add descriptions to variables
## Background

Typical Terraform pattern is to make use of the `description` field on variables to give context what they are used for. The `terraform.tfvars` file seems to have these descriptions already, but not for the variables themselves in `variables.tf`.

## Desired Change

Copy the description of variables from `terraform.tfvars` to `variables.tf`.

### PR DESCRIPTION
**Pull request by phretor**
to: @airbnb/binaryalert-maintainers

resolves #145 
resolves #173 
resolves #167
resolves #166 
resolves #154
resolves #153 
resolves #152 
resolves #142 
resolves #144

## Background

I see that there are many pending PR and open issues, so I tried to take the most important into account into one, comprehensive PR.

## Changes

* TF upgrade to latest version 0.14.5
* Python deps upgrades (at some point I suggest to migrate to Poetry as opposed to PIP)
* Added TF variable `aws_account_name` to solve #145 
* Changed `/usr/bin/env python3` into `/usr/bin/env python` to be venv friendly (Py3 should be the default)

## Testing

`python manage.py unit_test`

### CODE REVIEW

File: terraform/modules/lambda/versions.tf [hunk 1]
```diff
@@ -1,4 +1,4 @@
 
 terraform {
-  required_version = "~> 0.12.9"
+  required_version = ">= 0.13"
```
Comment (ryandeivert) [inline terraform/modules/lambda/versions.tf]: as of 0.14 and above this should include a `required_providers` block like:

```hcl
terraform {
  required_version = ">= 0.13"
  required_providers {
    aws = {
      source = "hashicorp/aws"
    }
  }
}
```

File: terraform/versions.tf [hunk 2]
```diff
@@ -1,3 +1,8 @@
 terraform {
-  required_version = "~> 0.12.9"
-}
\ No newline at end of file
+  required_version = ">= 0.13"
```
Comment (ryandeivert) [inline terraform/versions.tf:2]: the pr desc mentions tf 0.14 but 0.13 is pinned here. any reason?

**Review by ryandeivert — APPROVED** [review]
a few comments but lgtm

File: terraform/versions.tf (hunk 2 above)
Comment (phretor) [reply terraform/versions.tf:2]: Whoops, I missed that, but the automatic migration tool nor the validator did not complain: that's why it slipped. I can fix this in this PR or in a separate one. However you prefer.

File: terraform/modules/lambda/versions.tf (hunk 1 above)
Comment (phretor) [reply terraform/modules/lambda/versions.tf]: Okay, this is definitely to be handled.

**Comment by phretor** [comment]
I checked the logs of the CI job but I can't see any meaningful error message. I see that both jobs on Py 3.6 and Py 3.7 have errored, but I can't see the cause. Maybe you have some hints on how to dig deeper @ryandeivert ? Thanks!

File: docs/source/conf.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python
```
Comment (austinbyers) [inline docs/source/conf.py:1]: If I remember right, this is an auto-generated file; did it change with the latest version of `sphinx`?

File: terraform/modules/lambda/main.tf [hunk 4]
```diff
@@ -47,7 +47,7 @@ resource "aws_lambda_function" "function" {
   description   = var.description
   handler       = var.handler
   role          = aws_iam_role.role[0].arn
-  runtime       = "python3.6"
+  runtime       = "python3.7"
```
Comment (austinbyers) [inline terraform/modules/lambda/main.tf:50]: AWS Lambda has [Python3.8](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html) available, should we use that?

**Review by austinbyers — APPROVED** [review]
Awesome, thanks for the contribution! 

> but I can't see the cause

You can't see this page? https://travis-ci.org/github/airbnb/binaryalert/jobs/756818009

You should be able to click "details" on the X shown on the commit, but maybe it's not public for some reason

```
$ python --version
Python 3.6.7
$ pip --version
pip 20.1.1 from /home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/pip (python 3.6)
install
37.21s$ pip3 install -r requirements.txt
0.28s$ tests/ci_tests.sh
~~~~~~~~~~ [1] Bandit Security Linting ~~~~~~~~~~
[main]	INFO	Found project level .bandit file: ./.bandit
[main]	INFO	Using command line arg for excluded paths
[main]	INFO	Using ini file for skipped tests
[main]	INFO	Using command line arg for selected targets
[main]	INFO	Using command line arg for recursive scan
[main]	INFO	Using command line arg for aggregate output type
[main]	INFO	Using command line arg for max code lines output for issue
[main]	INFO	Using command line arg for severity level
[main]	INFO	Using command line arg for confidence level
[main]	INFO	Using command line arg for output format
[main]	INFO	Using command line arg for output file
[main]	INFO	profile include tests: None
[main]	INFO	profile exclude tests: None
[main]	INFO	cli include tests: None
[main]	INFO	cli exclude tests: B303,B322,B404,B603,B607
[main]	ERROR	Unknown test found in profile: B322
The command "tests/ci_tests.sh" exited with 2.
```

Looks like the CI environment will need to be updated with the newer version of Python, maybe? You can configure CI with the `.travis.yml` file


----

The real error is probably just the exclusion shown here:

```
[main]	INFO	cli exclude tests: B303,B322,B404,B603,B607
[main]	ERROR	Unknown test found in profile: B322
```
